src/yunohost/app.py

```diff
@@ -908,77 +908,101 @@ def confirm_install(confirm):
-                operation_logger_remove = OperationLogger('remove_on_failed_install',
-                                                          [('app', app_instance_name)],
-                                                          env=env_dict_remove)
-                operation_logger_remove.start()
+            # Execute remove script
+            operation_logger_remove = OperationLogger('remove_on_failed_install',
+                                                      [('app', app_instance_name)],
+                                                      env=env_dict_remove)
+            operation_logger_remove.start()
 
+            # Try to remove the app
+            try:
                 remove_retcode = hook_exec(
                     os.path.join(extracted_app_folder, 'scripts/remove'),
                     args=[app_instance_name], env=env_dict_remove
                 )[0]
-                # Remove all permission in LDAP
-                result = ldap.search(base='ou=permission,dc=yunohost,dc=org',
-                                    filter='(&(objectclass=permissionYnh)(cn=*.%s))' % app_instance_name, attrs=['cn'])
-                permission_list = [p['cn'][0] for p in result]
-                for l in permission_list:
-                    permission_remove(app_instance_name, l.split('.')[0], force=True)
-
-                if remove_retcode != 0:
-                    msg = m18n.n('app_not_properly_removed',
-                                 app=app_instance_name)
-                    logger.warning(msg)
-                    operation_logger_remove.error(msg)
+            # Here again, calling hook_exec could failed miserably, or get
+            # manually interrupted (by mistake or because script was stuck)
+            # In that case we still want to proceed with the rest of the
+            # removal (permissions, /etc/yunohost/apps/{app} ...)
+            except (KeyboardInterrupt, EOFError, Exception):
+                remove_retcode = -1
+                import traceback
+                logger.exception(m18n.n('unexpected_error', error=u"\n" + traceback.format_exc()))
+
+            # Remove all permission in LDAP
+            result = ldap.search(base='ou=permission,dc=yunohost,dc=org',
+                                filter='(&(objectclass=permissionYnh)(cn=*.%s))' % app_instance_name, attrs=['cn'])
+            permission_list = [p['cn'][0] for p in result]
+            for l in permission_list:
+                permission_remove(app_instance_name, l.split('.')[0], force=True)
+
+            if remove_retcode != 0:
+                msg = m18n.n('app_not_properly_removed',
+                             app=app_instance_name)
+                logger.warning(msg)
+                operation_logger_remove.error(msg)
+            else:
+                try:
+                    _assert_system_is_sane_for_app(manifest, "post")
+                except Exception as e:
+                    operation_logger_remove.error(e)
```

Comment:
Again, is the error message good UX here? 

Like informing the user about a list of things that needs to be fixed seems like a good idea '-'

Reply:
Yes we just reuse the message from https://github.com/YunoHost/yunohost/blob/stretch-unstable/src/yunohost/app.py#L2982